File: Examples/ClassDiagram/TypeScript/Player.ts
import { Library } from "./Library";
import { Playable } from "./Playable";

/** Drives playback over a library. Depends on Library and the Playable interface. */
export class Player {
    private nowPlaying: Playable | null = null;

    constructor(private readonly library: Library) {}

    play(item: Playable): void {
        this.nowPlaying = item;
        item.play();
    }

    playFirstItem(playlistNamed: string): void {
        const item = this.library.playlist(playlistNamed)?.items[0];
        if (item) {
            this.play(item);
        }
    }
}

File: Examples/ClassDiagram/Dart/player.dart
import 'library.dart';
import 'playable.dart';

/// Drives playback over a library. Depends on [Library] and the [Playable] interface.
class Player {
  Player(this.library);

  final Library library;
  Playable? nowPlaying;

  void play(Playable item) {
    nowPlaying = item;
    item.play();
  }

  void playFirstItem(String playlistNamed) {
    final items = library.playlist(playlistNamed)?.items;
    if (items != null && items.isNotEmpty) {
      play(items.first);
    }
  }
}
